data_loader.py

```diff
@@ -28,45 +31,103 @@ def disconnect(self):
+            " ", "_"
+        ).str.lower()
+        review_status["article_id"] = review_status["article_id"].apply(
+            lambda x: x.strip()
+        )
+        date_completed["reference_id"] = date_completed["reference_id"].values.astype(
+            str
+        )
+
+        for article in list_of_articles:
+            excel_dict = {}
+            article_status = review_status.loc[
+                review_status["article_id"] == article["_id"]
+            ]
+            if len(article_status) > 1:
+                excel_dict["status.duplicates"] = len(article_status)
+                excel_dict["status.duplicates.reference_id"] = article_status[
+                    "reference_id"
+                ].tolist()
+
+            if article_status.empty:
+                excel_dict.update(article_status.to_dict())
+                reference_id = None
+            else:
+                excel_dict.update(article_status.tail(1).to_dict("records")[0])
+                reference_id = article_status["reference_id"].values
+
+            article_date = date_completed.loc[
+                date_completed["reference_id"].isin(reference_id)
+            ]
+
+            if article_date.empty:
+                excel_dict.update(article_date.to_dict())
+            else:
+                excel_dict.update(article_date.tail(1).to_dict("records")[0])
+
+            article.update(excel_dict)
+
+        return list_of_articles
+
     def getDataBatch(self, batch_size):
         """
         Generator - Get data from source on batches.
 
         :returns One list for each batch. Each of those is a list of
                  dictionaries with the defined rows.
         """
-        # TODO: implement - this dummy implementation returns one batch of data
-        yield [
-            {
-                "headline.main": "The main headline",
-                "_id": "1234",
-            }
+        with open(self.args.api_response_file) as json_file:
+            data = json.load(json_file)
+
+        list_of_articles = [
+            {k: v for k, v in self._flatten_dict(article)}
+            for article in self._combine_excel(data["response"]["docs"])
         ]
+        for i in range(0, len(list_of_articles), batch_size):
+            yield list_of_articles[i : i + batch_size]
 
     def getSchema(self):
         """
         Return the schema of the dataset
         :returns a List containing the names of the columns retrieved from the
         source
         """
-        # TODO: Update the schema to reflect the item (row)
-        schema = [
-            "title",
-            "body",
-            "created_at",
-            "id",
-            "summary",
-            "abstract",
-            "keywords",
-        ]
-
-        return schema
+        example_result = self.getDataBatch(10)
+        all_keys = set().union(*(d.keys() for batch in example_result for d in batch))
+        return all_keys
```

Comment:
```suggestion
        batch = next(self.getDataBatch(10), ())
        fields = set()
        for entry in batch:
            fields |= entry.keys()
        return sorted(fields)
```
IMO much more readable. Also, it is supposed to return a list and not set :)